posthog/api/routing.py

```diff
@@ -47,11 +48,15 @@ def team(self) -> Team:
         if team_from_token:
             return team_from_token
 
+        user = cast(User, self.request.user)
         if self.legacy_team_compatibility:
-            team = self.request.user.team
+            team = user.team
             assert team is not None
             return team
-        return Team.objects.get(id=self.team_id)
+        try:
+            return Team.objects.get(id=self.team_id)
```

Comment:
Why are we doing this extra check when you could just return `user.team`, FK will already protect that the project exists.

Reply:
Got it, makes sense!